Tests/kaas/plugin/plugin_cluster_stacks.py

```diff
@@ -0,0 +1,366 @@
+    self.cs_class_name = f"openstack-{self.cs_name}-{str(self.cs_k8s_version).replace('.', '-')}-{self.cs_version}"
+    self.cs_namespace = self.config.get("cs_namespace", "default")
+    self.cs_pod_cidr = self.config.get('cs_pod_cidr', '192.168.0.0/16')
+    self.cs_service_cidr = self.config.get('cs_service_cidr', '10.96.0.0/12')
+    self.cs_external_id = self.config.get('cs_external_id', 'ebfe5546-f09f-4f42-ab54-094e457d42ec')
+    self.cs_k8s_patch_version = self.config.get('cs_k8s_patch_version', '6')
+
+    if not self.clouds_yaml_path:
+        raise ValueError("CLOUDS_YAML_PATH environment variable not set.")
+
+    required_env = {
+        'CLUSTER_TOPOLOGY': 'true',
+        'EXP_CLUSTER_RESOURCE_SET': 'true',
+        'EXP_RUNTIME_SDK': 'true',
+        'CS_NAME': self.cs_name,
+        'CS_K8S_VERSION': self.cs_k8s_version,
+        'CS_VERSION': self.cs_version,
+        'CS_CHANNEL': self.cs_channel,
+        'CS_CLOUDNAME': self.cs_cloudname,
+        'CS_SECRETNAME': self.cs_secretname,
+        'CS_CLASS_NAME': self.cs_class_name,
+        'CS_NAMESPACE': self.cs_namespace,
+        'CS_POD_CIDR': self.cs_pod_cidr,
+        'CS_SERVICE_CIDR': self.cs_service_cidr,
+        'CS_EXTERNAL_ID': self.cs_external_id,
+        'CS_K8S_PATCH_VERSION': self.cs_k8s_patch_version,
+        'CS_CLUSTER_NAME': self.cluster_name,
+    }
+    # Update the environment variables
+    os.environ.update({key: str(value) for key, value in required_env.items()})
+
+
+def setup_git_env(self):
+    # Setup Git environment variables
+    git_provider = self.config.get('git_provider', 'github')
+    git_org_name = self.config.get('git_org_name', 'SovereignCloudStack')
+    git_repo_name = self.config.get('git_repo_name', 'cluster-stacks')
+
+    os.environ.update({
+        'GIT_PROVIDER_B64': base64.b64encode(git_provider.encode()).decode('utf-8'),
+        'GIT_ORG_NAME_B64': base64.b64encode(git_org_name.encode()).decode('utf-8'),
+        'GIT_REPOSITORY_NAME_B64': base64.b64encode(git_repo_name.encode()).decode('utf-8')
+    })
+
+    git_access_token = os.getenv('GIT_ACCESS_TOKEN')
+    if git_access_token:
+        os.environ['GIT_ACCESS_TOKEN_B64'] = base64.b64encode(git_access_token.encode()).decode('utf-8')
+    else:
+        raise ValueError("GIT_ACCESS_TOKEN environment variable not set.")
+
+
+class PluginClusterStacks(KubernetesClusterPlugin):
+    def __init__(self, config_file=None):
+        self.config = load_config(config_file) if config_file else {}
+        logger.debug(self.config)
+        self.working_directory = os.getcwd()
+        logger.debug(f"Working from {self.working_directory}")
+
+    def create_cluster(self, cluster_name="scs-cluster", version=None, kubeconfig_filepath=None):
+        self.cluster_name = cluster_name
+        self.cluster_version = version
+
+        # Setup variables
+        setup_environment_variables(self)
+        setup_git_env(self)
```

Comment:
I don't think we should use `os.environ.update` to update the environment of this process. We should construct the dict with the updates that we need and then pass this dict to `subprocess.run` (well possible with `self._run_subprocess`).